scripts/package

```diff
@@ -39,7 +39,44 @@ do
     docker pull "$BASE_IMAGE"
 done
 
-docker build --build-arg ARCH=${ARCH} -t ${IMAGE} -f package/Dockerfile .
+function convert_version_to_major_minor_x() {
+    local version="$1"
+    if [[ "$version" =~ ^v([0-9]+)\.([0-9]+)\. ]]; then
+        echo "v${BASH_REMATCH[1]}.${BASH_REMATCH[2]}.x"
+    else
+        echo "Invalid version format: $version"
+    fi
+}
+
+function get_branch() {
+  local version_file="version"
+  if [[ ! -f $version_file ]]; then
+    echo "Error: Version file '$version_file' not found."
+    exit 1
+  fi
+
+  local version=$(cat "$version_file")
+  local branch=$(convert_version_to_major_minor_x "$version")
+
+  echo "${branch}"
+}
+
+BRANCH=$(get_branch)
+
+wget -q "https://raw.githubusercontent.com/longhorn/dep-versions/refs/heads/${BRANCH}/versions.json" -O package/versions.json || {
+  echo "Failed to fetch from branch $BRANCH, trying main branch..."
+  wget -q "https://raw.githubusercontent.com/longhorn/dep-versions/refs/heads/main/versions.json" -O package/versions.json || {
```

Comment:
In longhorn-manager/master branch, `get_branch` will return "v1.9.x", a branch that does not exist in longhorn-manager or in dep-versions repo.  Is that why there is a failover to dep-version/head branch here?  It will also use dep-versions/main to build backport branches of longhorn-manager until the dep-version branches are created with their own versions.json.  (Which will be pretty soon, I would guess.)

Reply:
yes, you are right!